### DISCUSSION TITLE
Issue with parameter estimation involving subject status (categorical covariate)

### DISCUSSION THREAD

**Discussion by nlmixr-user** [nlmixr2/nlmixr2#93] [Q&A]
Hi, 
I would like to add a food effect on the absorption rate. I created a column "FOOD" in my dataset with 0 and 1 as modalities. 
In the "ini" block, the initial estimates were listed as follow:
tkafa (for fasted state) and tkafe (for fed state) 
The alternatives tried in the "model block" and their respective results are:
case 1:
if (FOOD ==0) {
ka = exp(tkafa+eta.ka)
} else {
ka = exp(tkafe + eta.ka)
};
Result : 
![param-estimation1](https://user-images.githubusercontent.com/120671995/208126698-0b63ac14-ef24-4578-bf1b-076a65a6fbd2.png)
case 2: 
ka = exp(tkafe+eta.ka)*FOOD + exp(tkafa+eta.ka)
Result:
![param-estimation2](https://user-images.githubusercontent.com/120671995/208124809-34e85108-96eb-478d-a479-57452dcb5368.png)
case 3:
ka = exp(tkafe*FOOD+tkafa+eta.ka);
Result:
![param-estimation3](https://user-images.githubusercontent.com/120671995/208125832-e64e308e-273c-49bb-b109-772cf0c020da.png)
The population parameters are either not estimated or not back-transformed or both. I'm using focei estimation method. Is there a way to add a categorical effect on a parameter in nlmixr? Thank you in advance.

**Comment by mattfidler** [comment]
The methods you use above are generally non-identifiable (which means there is a variety of parameter estimates that give the same results)

The standard way to add a mu-referenced covariate is:

```R
ka <- exp(tka+etaka+FOOD*food_cov)
```

**Reply by nlmixr-user** [reply]
Hi Matthew, 
Thank you for the quick feedback. I tried the mu-referencing with both estimation methods (saem & focei) and this was quite similar to what I obtained previously with other alternatives. 
Using the saem estimation method, the results were very bad for all parameters and seems that focei worked well for individual parameters estimation but not for population parameters. Please find below the code used: 
![image](https://user-images.githubusercontent.com/120671995/208664432-85bcdf04-9314-4a9d-b706-a5ea34ec70c5.png)
Results:
![image](https://user-images.githubusercontent.com/120671995/208664595-b2d17ed4-065c-4c21-b8f6-e76794dca9c4.png)

Thank you so much for your help.

**Comment by mattfidler** [comment]
In `saem` this turns into a linear for these parameters model allowing more accurate identification of `food_cov` (mabye someday in focei too).

While there are other ways, this is the way I suggest doing this binary response.

If there is more than one category, then  you would create indicators for each of the categories (except the last) and then use `tka+etaka+cat1*cov1+cat2*cov2+...)`

**Comment by mattfidler** [comment]
I converted this to a discussion, since it is more of a Q&A sort of item.
